snapcraft/cli/lifecycle.py

```diff
@@ -23,6 +24,14 @@
 from . import env
 
 
+def _reexec_as_root():
+    click.echo('Reexecuting the build as root')
+    try:
+        subprocess.check_call(['sudo'] + sys.argv)
+    except subprocess.CalledProcessError:
+        sys.exit(1)
```

Comment:
Why are you catching this exception here, instead of letting the caller handle it?
It already has a sys.exit(1), plus echo of the error.